feat:  Groups Management documentation (#125)
mia-platform/documentation: docs/development_suite/identity-and-access-management/group-management.md

title: Manage Groups
sidebar_label: Manage Groups
---

Mia-Platform Console allows you to assign and manage authorization privileges to groups of users at Company level. Groups aim to simplify the management of a large number of users who have to be assigned the same privileges (e.g. a work team).

Groups are considered a type of [Identity](/development_suite/identity-and-access-management/overview.md#identity-and-access-management) whereby Users with enough permissions can assign and manage roles on the following resources:

* Company
* Project
* Runtime Environment

:::note
To find out more about Roles management, check out the available [Capabilities](/development_suite/identity-and-access-management/console-levels-and-permission-management.md#users-capabilities-inside-console) that can be assigned to an Identity.
:::

## Managing Company Groups

Each Company can have its own Groups. These types of Identities are configurable and manageable exclusively by a Company Owner from the dedicated Groups section within the Company Overview. 

At Company level, for each Group, it is possible to manage:

* Name: the name assigned to the Group
* Role: the Group Company Role

> * Members: adding and/or removing Users from the Group

Author: ```suggestion
* Members: adding and/or removing users from the Group
```

In the case that a User has access to more Company Groups, or has a specific individual Company role assigned, the removal from that Group will NOT result in the loss of access to the Company for that User; for this reason, it is necessary to pay close attention to what privileges are still left to that User on the Company resources. 
:::

<!-- 
<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

TODO: ![Edit group](./img/group-management/edit_group.png)

  </div>
</div> -->

### Deleting a Group

A Group can be deleted by accessing its detail page and clicking on the *Delete* button at the end of the page and then confirming the action.

:::caution
Group deletion may result in loss of access to the Company for those Group members who are not members of at least one other Company Group or who do not have a specific individual Company role assigned.   

In the case that a Group member has access to the Company solely from its membership of the deleted Group, the removal from that Group will result in the loss of access to the Company for that User.  

> In the case that a User has access to more Company Groups, or has a specific individual Company role assigned, the deletion of the Group will NOT result in loss of access to the Company for that User; for this reason, it is necessary to pay attention to what privileges are still left to that User on the Company resources.

Author: This concept is repeated here and for group member removal

Reviewer: I thought it was useful to reiterate because both actions have an impact on that topic, so if a person were to read only one of the two paragraphs, they would find that specific information

---
id: group-management
title: Manage Groups
sidebar_label: Manage Groups
---

> Mia-Platform Console allows to assign and manage authorization privileges to groups of users at Company level. Groups aim at simplify the management a large number of users who have to be assigned the same privileges (e.g. a work team).

Reviewer 2: ```suggestion
Mia-Platform Console allows you to assign and manage authorization privileges to groups of users at Company level. Groups aim to simplify the management of a large number of users who have to be assigned the same privileges (e.g. a work team).
```

<!-- 

ITA version: 
Mia-Platform Console permette di assegnare e gestire privilegi di autorizzazione a gruppi di utenti presenti o meno a livello di Company. In questo modo risulta più agevole gestire un grande numero di utenti a cui dover assegnare gli stessi privilegi (es. work team).

> Mia-Platform Console enables certain Users to control other identities ability to access resources. This grants them ownership over the decision-making process regarding which resources are accessible and at what level of authorization. -->

Reviewer 2: perhaps too generic

Reviewer: it was a comment, it has been removed

Groups is considered a type of [Identity](/development_suite/identity-and-access-management/overview.md#identity-and-access-management) to which Users with enough permissions can assign and manage roles on the following resources:

* Company
* Project
* Runtime Environment

:::note
To find out more about Roles management, check out the available [Capabilities](/development_suite/identity-and-access-management/console-levels-and-permission-management.md#users-capabilities-inside-console) that can be assigned to an Identity.
:::

<!-- In particular, a User can be assigned a Role on the following resources:

* Company
* Project
* Runtime Environment -->

<!-- Only Users with enough privileges will be granted access to the Identities administration portal and will be able to change identity Roles. -->

## Managing Company Groups

> <!-- A User with enough administrative permission on a specific Company will be able to view the existing identities in the Company, add new ones and change the existing identities' Roles. -->

Reviewer 2: ```suggestion
<!-- A User with enough administrative permission on a specific Company will be able to view the existing Identities in the Company, change their roles and add new Identities. -->
```

> Each Company has its own Groups, which are configurable and manageable solely by Company Owner from the dedicated Groups section inside the Company Overview. 

Reviewer 2: ```suggestion
Each Company can have its own Groups. These types of Identities are configurable and manageable exclusively by a Company Owner from the dedicated Groups section within the Company Overview. 
```

At Company level, for each Group, it is possible to manage:

* Name: the name assigned to the Group
* Role: the Group Company Role
* Members: adding and/or removing Users from the Group

> Existing Groups can also be found within the Identities portal at both Company and Project level, where it is possible to manage Roles assigned to them as well as for Users and Service Account Users. 

Reviewer 2: ```suggestion
Existing Groups can also be found within the Identities portal at both Company and Project level, where it is possible to manage Roles assigned to them as well as for the other Identity types. 
```

<!-- TODO: ![Group table](./img/group-management/group_table.png) -->

:::caution

> Please note that some permissions defined by the Company Role may be inherited on the Projects and Runtime Environment owned by the Company itself.  

Reviewer 2: ```suggestion
Please note that some permissions defined by the Company Role may be inherited on the Projects and Runtime Environments of the Company itself.  
```

> Always pay attention when assigning Roles in order to avoid providing undesired access to resources!

Reviewer 2: it may be helpful to put also here a link to the Console levels and permission management page

:::

### Creating a Group

<!-- The Company Owner can add a new User by pressing the *Add user* button. The User invitation process will require the invited User email to be provided. -->

> <!-- Si può creare un gruppo utilizzando il bottone *Add User* e nel processo di creazione si possono definire il suo nome, il suo ruolo sulla Company e tutti i suoi membri. I membri aggiunti al gruppo possono anche non appartenere alla Company (cioè non hanno mai avuto un ruolo come Identities sulla Company in questione) e in tal caso viene richiesta la loro email. -->

Reviewer 2: should be removed

Reviewer: removed

> The Company Owner can start the group creation process by pressing *Create group* button inside the Groups section. In the creation process it is required to define the Group name and the Company role to assign. It is also possible to add members to the Group.

Reviewer 2: ```suggestion
The Company Owner can start the group creation process by pressing the *Create group* button inside the Groups section. In the creation process it is required to define the Group name and the Company role to assign. It is also possible to add members to the Group.
```

:::info
During the creation process, the Group can be also created without members added to it. This action can be performed even after the Group creation.
:::

<!-- <div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

TODO: ![Add group](./img/group-management/add_user.png)

  </div>
</div> -->

> It is possible to add to a Group both Users who already have access to the Company and new Users who will consequently be given access to the Company. For the latter, it is necessary to insert the e-mail address to which the invitation to join the Company will be sent.

Reviewer 2: ```suggestion
It is possible to add to a Group both Users who already have access to the Company and new Users who do not have access to the Company (but which already exist on the Platform). For the latter, once inserted their e-mail address, they will receive an invitation to join that Group of the Company.  As long as they are part of at least one Group of a Company, they can have access to the resources of that Company.
```

> The provided email will be searched throughout existing users on the Platform and if a match is found the user will be invited to the Company with the defined Role.

Reviewer 2: ```suggestion
The provided emails will be searched throughout existing users on the Platform and, if a match is found, the user will be invited to the Group with the defined Role of that specific Group.
```

:::info

> If the User does not exist in the Platform, please open a Service Request for it to be invited to registering in the Platform.

Reviewer 2: ```suggestion
If the User does not exist in the Platform, please open a Service Request to invite them to register in the Platform.
```

:::

### Editing a Group

> After having created a Group, it is possible to enter its detail page.

Reviewer 2: ```suggestion
After the creation of a Group, it is possible to enter its detail page.
```

Once entered, a Company Owner can perform the following actions:

> * Changing Group name
> * Editing Group Company role
> * Adding new members to the Group
> * Removing members to the Group

Reviewer 2: ```suggestion
* Change Group name
* Edit Group Company role
* Add new members to the Group
* Remove members from the Group
```

:::info
In the case that a User has access to the Company solely from its membership of that Group, the removal from that Group will result in loss of access to the Company for that User.  

> In the case that a User has access even to other Company Groups, or has a specific Company role assigned, the removal from that Group will NOT result in loss of access to the Company for that User; however, it is necessary to pay attention to what privileges are still left to that User on the Company resources. 

Reviewer 2: ```suggestion
In the case that a User has access to more Company Groups, or has a specific individual Company role assigned, the removal from that Group will NOT result in the loss of access to the Company for that User; For this reason, it is necessary to pay close attention to what privileges are still left to that User on the Company resources. 
```

:::

> <!-- A User Role in the Company can be modified: to do so, simply click on the edit button for the desired User row and select the new Role. -->

Reviewer 2: should be removed

<!-- 
<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

TODO: ![Edit group](./img/group-management/edit_group.png)

  </div>
</div> -->

> ### Removing a Group

Reviewer 2: ```suggestion
### Deleting a Group
```

> A Group can be deleted by accessing its detail page and clicking on the *Delete* button at the bottom of the page and confirming the action.

Reviewer 2: ```suggestion
A Group can be deleted by accessing its detail page and clicking on the *Delete* button at the end of the page and then confirming the action.
```

> :::info

Reviewer 2: may be a warning banner

> The deletion of the Group may implies the loss of access to the Company for some Group members.  

Reviewer 2: ```suggestion
Group deletion may result in loss of access to the Company for those Group members who are not members of at least one other Company Group or who do not have a specific company role assigned.   
```

> Group deletion may result in loss of access to the Company for those Group members who are not members of at least one other Company Group or who do not have a specific company role assigned.   

Reviewer 2: ```suggestion
Group deletion may result in loss of access to the Company for those Group members who are not members of at least one other Company Group or who do not have a specific individual Company role assigned.   
```

> In the case that a group member has access to the Company solely from its membership of the deleted Group, the removal from that Group will result in loss of access to the Company for that User.  

Reviewer 2: ```suggestion
In the case that a Group member has access to the Company solely from its membership of the deleted Group, the removal from that Group will result in the loss of access to the Company for that User.  
```

> In the case that a User has access even to other Company Groups, or has a specific Company role assigned, the deletion of the Group will NOT result in loss of access to the Company for that User; however, it is necessary to pay attention to what privileges are still left to that User on the Company resources.

Reviewer 2: ```suggestion
In the case that a User has access to more Company Groups, or has a specific individual Company role assigned, the deletion of the Group will NOT result in loss of access to the Company for that User; for this reason, it is necessary to pay attention to what privileges are still left to that User on the Company resources.
```

:::

> <!-- A User can be removed from the Company by clicking the trash icon on the table and confirming the action. -->

Reviewer 2: should be removed

Reviewer: yeah it has been removed

<!-- 
<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

TODO: ![Delete group](./img/group-management/delete_group.png)

  </div>
</div> -->

<!-- :::warning

> Removing a User from the Company will kick the User out of every Project and Runtime Environment. While the User may be invited back in the Company, all their previously existing Roles will be lost and cannot be recovered, meaning they must be reassigned from scratch.

Reviewer: yeah it has been removed

:::caution
Please note that some permissions granted by the assigned Company role may be inherited on the Projects and Runtime Environments of the Company itself.  
Always pay attention when assigning roles in order to avoid providing undesired access to resources!
You can check which capabilities are granted according to the chosen role by visiting the [Console Levels and Permission Management](/development_suite/identity-and-access-management/console-levels-and-permission-management.md#users-capabilities-inside-console) documentation page.
:::

### Creating a Group

The Company Owner can start the group creation process by pressing the *Create group* button inside the Groups section. In the creation process it is required to define both the name and Company role of the group. It is also possible to optionally add members to the group.

:::info
During the creation process, the group can be also created without members added to it. This action can be performed even after the group creation.
:::

<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

![Add group](./img/group-management/add_group.png)

  </div>
</div>

> Both users who are already part of the Company and new users whose have no access yet, can be added to a group.

Reviewer 2: ```suggestion
Both users who are already part of the Company and new users whose have no access yet can be added to a group.
```

For the latter, the user must already be registered on the Platform. These users will receive an invitation to join the Company group.
As long as a user is part of at least one group of a Company, this has access to the resources of that Company.
:::info

> If the user does not exist in the Platform, please open a Service Request to invite it to register in the Platform.

Reviewer 2: ```suggestion
If the user does not exist on the Platform, please open a Service Request to invite it to register in the Platform.
```

:::

<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

![Edit group](./img/group-management/edit_group.png)

  </div>
</div>

### Managing Group role at Company level

Groups are shown into the [Identities portal](/development_suite/identity-and-access-management/overview.md#identities-portal) inside the Company Overview.
As a result, the Company role of a group can be modified not only from the Groups section, but also from the Identities portal: to do so, simply click on the edit button for the desired group row and select the new role.

<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

![Edit group role at Company level](./img/group-management/edit-group-role-at-company-level.png)

  </div>
</div>

### Managing Group role at Project level

Groups which have access to a Company are shown into the [Identities portal](/development_suite/identity-and-access-management/overview.md#identities-portal) inside the Project Settings Area of a Project belonging to that specific Company.

> Group role in the Project or any of the Project Runtime Environments can be modified. To do so, just open the editing dialog and select the proper role for the Project itself or for each Runtime Environment.

Reviewer 2: ```suggestion
You can change the role of the group in the Project or in any of the Project Runtime Environments. To do so, just open the editing dialog and select the new role of the group for the Project itself and/or for Project Runtime Environments.
```

<div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

![Edit group role at Project level](./img/group-management/edit-group-role-at-project-level.png)

  </div>
</div>

:::note

> Even though the Project Identities administration portal shows all the Company identities, this does not mean that all the identities have access to the Project, as this depends on their assigned role in the Company and how the permissions are inherited.

Reviewer 2: ```suggestion
Even though the Project Identities administration portal shows all the Company identities, this does not mean that all the identities have also access to the Projects, as this depends on their assigned role in the Company and how the permissions are inherited.
```

For further information about permissions and role inheritance, check out the [Console Levels and Permission Management](/development_suite/identity-and-access-management/console-levels-and-permission-management.md) page.
:::

### Deleting a Group

A group can be deleted by accessing its detail page and clicking on the *Delete* button at the end of the page and then confirming the action.

:::caution
Group deletion may result in loss of access to the Company for those group members who are not members of at least one other Company group or who do not have a specific individual Company role assigned.   

> In the case that a group member has access to the Company solely from its membership of the deleted group, the removal from that group will result in the loss of access to the Company for that user.  

Reviewer 2: ```suggestion
In the event that a member of a group has access to the Company solely by virtue of being a member of a group, the deletion of that group or the removal from it will result in the loss of access to the Company for that user.  
```

Please note that some permissions granted by the assigned Company role may be inherited on the Projects and Runtime Environments of the Company itself.  
Always pay attention when assigning roles in order to avoid providing undesired access to resources!
You can check which capabilities are granted according to the chosen role by visiting the [Console Levels and Permission Management](/development_suite/identity-and-access-management/console-levels-and-permission-management.md#users-capabilities-inside-console) documentation page.
:::

### Creating a Group

The Company Owner can start the group creation process by pressing the *Create group* button inside the Groups section. In the creation process it is required to define both the name and Company role of the group. It is also possible to optionally add members to the group.

:::info
During the creation process, the group can be also created without members added to it. This action can be performed even after the group creation.
:::

<!-- <div style={{display: 'flex', justifyContent: 'center'}}>
  <div style={{display: 'flex', width: '600px'}}>

TODO: ![Add group](./img/group-management/add_user.png)

  </div>
</div> -->

> It is possible to add to a group both users who already have access to the Company and new users who do not have access to the Company (but which already exist on the Platform). For the latter, once inserted their email address, they will receive an invitation to join that group of the Company.  As long as they are part of at least one group of a Company, they can have access to the resources of that Company.

Reviewer 3: ```suggestion
Both users who are already part of the Company and new users whose have no access yet, can be added to a group.
For the latter, the user must already be registered on the Platform. These users will receive an invitation to join the Company group.
As long as a user is part of at least one group of a Company, this has access to the resources of that Company.
```

> 
> The provided emails will be searched throughout existing users on the Platform and, if a match is found, the user will be invited to the group with the defined tole of that specific group.
> 

Reviewer 3: ```suggestion

```

:::info
If the user does not exist in the Platform, please open a Service Request to invite it to register in the Platform.
:::

### Editing a Group

> After the creation of a group, it is possible to enter its detail page.
> Once entered, a Company Owner can perform the following actions:

Reviewer 3: ```suggestion
After a group has been created, it is possible to view its detail page.
From this page, a Company Owner can perform the following actions:
```

  </div>
</div> -->

### Managing group Company role

Groups are shown into the [Identities portal](/development_suite/identity-and-access-management/overview.md#identities-portal) inside the Company Overview.

> Company role of a group can be modified: to do so, simply click on the edit button for the desired group row and select the new role.

Reviewer 2: maybe it's better to clarify that you can modify the comapny role of the group also in the Identities section?
